internal/services/authorization/role_management_policy.go

```diff
@@ -0,0 +1,634 @@
+					return nil, fmt.Errorf("require_approval is true, but no approval_stages are provided")
+				}
+			}
+
+			var approvalReqd bool
+			var approvalStages []map[string]interface{}
+
+			if settingsRaw, ok := approvalEndUserAssignment.Values["setting"]; ok {
+				settings := settingsRaw.(map[string]interface{})
+
+				if approvalReqdRaw, ok := settings["isApprovalRequired"]; ok {
+					approvalReqd = approvalReqdRaw.(bool)
+				}
+
+				if len(model.ActivationRules) == 1 {
+					if approvalReqd != model.ActivationRules[0].RequireApproval {
+						approvalReqd = model.ActivationRules[0].RequireApproval
+					}
+				}
+
+				if metadata.ResourceData.HasChange("activation_rules.0.approval_stage") {
+					if len(model.ActivationRules) == 1 {
+						approvalStages = make([]map[string]interface{}, len(model.ActivationRules[0].ApprovalStages))
+						for i, stage := range model.ActivationRules[0].ApprovalStages {
+							primaryApprovers := make([]map[string]interface{}, len(stage.PrimaryApprovers))
+							for ia, approver := range stage.PrimaryApprovers {
+								primaryApprovers[ia] = map[string]interface{}{
+									"id":       approver.ID,
+									"userType": approver.Type,
+								}
+							}
+
+							approvalStages[i] = map[string]interface{}{
+								"PrimaryApprovers": primaryApprovers,
+							}
+						}
+					}
+				} else {
+					if approvalStagesRaw, ok := settings["approvalStages"]; ok {
+						approvalStages = approvalStagesRaw.([]map[string]interface{})
+					}
+				}
+			}
+
+			var id, ruleType string
+			var target map[string]interface{}
+			if idRaw, ok := approvalEndUserAssignment.Values["id"]; ok {
+				id = idRaw.(string)
+			}
+			if ruleTypeRaw, ok := approvalEndUserAssignment.Values["ruleType"]; ok {
+				ruleType = ruleTypeRaw.(string)
+			}
+			if targetRaw, ok := approvalEndUserAssignment.Values["target"]; ok {
+				target = targetRaw.(map[string]interface{})
+			}
+
+			updatedRules = append(updatedRules, map[string]interface{}{
+				"id":       id,
+				"ruleType": ruleType,
+				"target":   target,
+				"setting": map[string]interface{}{
+					"isApprovalRequired": approvalReqd,
+					"approvalStages":     approvalStages,
+				},
+			})
+		}
+	}
+
+	if metadata.ResourceData.HasChange("activation_rules.0.required_conditional_access_authentication_context") {
+		if authEndUserAssignment, ok := existingRules["AuthenticationContext_EndUser_Assignment"]; ok && authEndUserAssignment.Values != nil {
+			var claimValue string
+			if claimValueRaw, ok := authEndUserAssignment.Values["claimValue"]; ok {
+				claimValue = claimValueRaw.(string)
+			}
+
+			var isEnabled bool
+			if _, set := metadata.ResourceData.GetOk("activation_rules.0.required_conditional_access_authentication_context"); set {
+				isEnabled = true
+				if len(model.ActivationRules) == 1 {
+					claimValue = model.ActivationRules[0].RequireConditionalAccessContext
+				}
+			} else {
+				isEnabled = false
```

Comment:
We could simplify this by starting with isEnabled is false

Reply:
Good shout, will tidy this up 👍